pkg/controller/controller_utils.go

```diff
@@ -25,83 +26,97 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
 	"github.com/golang/glog"
 	"sync/atomic"
 )
 
 const (
 	CreatedByAnnotation = "kubernetes.io/created-by"
-	updateRetries       = 1
+
+	// If a watch drops a delete event for a pod, it'll take this long
+	// before a dormant rc waiting for those packets is woken up anyway. It is
+	// specifically targeted at the case where some problem prevents an update
+	// of expectations, without it the RC could stay asleep forever. This should
+	// be set based on the expected latency of watch events.
+	//
+	// Currently an rc can service (create *and* observe the watch events for said
+	// creation) about 10-20 pods a second, so it takes about 1 min to service
+	// 500 pods. Just creation is limited to 20qps, and watching happens with ~10-30s
+	// latency/pod at the scale of 3000 pods over 100 nodes.
```

Comment:
> watching happens with ~10-30s latency/pod at the scale of 3000 pods over 100 nodes

From the time you submit a create for a pod to when the same pod comes back down the watch channel is 10-30s. So if you create 3000 pods, all of them will come down the watch channel sometime between the 0->3m mark, depending on when in the 0->2.5m mark they were created. If you're asking why there's so much latency, that a different story alltogether. Generally if the watch latencies start going above 30s, something else breaks and you no longer care about the replica count.